src/watchdog/utils/dirsnapshot.py

```diff
@@ -229,7 +229,7 @@ def walk(root):
                 yield _
             if recursive:
                 for path, st in entries:
-                    if S_ISDIR(st.st_mode):
+                    if S_ISDIR(st.st_mode) and os.access(path, os.R_OK):
```

Comment:
I would rather use try/except here as described in the [documentation](https://docs.python.org/3/library/os.html#os.access). It will stick with the EAFP idiomatic: "it's easier to ask for forgiveness than permission" :)